### PR TITLE
Inertial particles and their google test

### PR DESCRIPTION
**Pull request by pakserep**
Inertial particles added with their google test for 3 different test cases.

### CODE REVIEW

**Comment by pakserep** [comment]
Hi @mmcgurn 
Could you please review this PR? it shows one of the checks is failing though! Should I do something to fix it? thanks.

**Comment by mmcgurn** [comment]
@pakserep It looks like there were a lot of changes between when you started working on the branch and now.  It won't let you merge in the code until you resolve.  I like to use git rebase https://docs.github.com/en/github/getting-started-with-github/using-git/about-git-rebase .  My favorite way to git rebase is to use GitHub Desktop rebating https://docs.github.com/en/desktop/contributing-and-collaborating-using-github-desktop/keeping-your-local-repository-in-sync-with-github/syncing-your-branch#rebasing-your-project-branch-onto-another-branch

File: ablateCore/particles/particleInertial.c [hunk 1]
```diff
@@ -0,0 +1,341 @@
...
+
+    ierr = DMSwarmDestroyGlobalVectorFromField(sdm, FluidVelocity, &fluidVelocity);CHKERRQ(ierr);
+    ierr = DMSwarmDestroyGlobalVectorFromField(sdm,DMSwarmPICField_coor, &particlePosition);CHKERRQ(ierr);
+    ierr = DMSwarmDestroyGlobalVectorFromField(sdm,ParticleVelocity, &particleVelocity);CHKERRQ(ierr);
+    ierr = DMSwarmDestroyGlobalVectorFromField(sdm,ParticleDiameter, &particleDiameter);CHKERRQ(ierr);
+    ierr = DMSwarmDestroyGlobalVectorFromField(sdm,ParticleDensity, &particleDensity);CHKERRQ(ierr);
+
+    PetscFunctionReturn(0);
+}
+
```
Comment (mmcgurn) [inline ablateCore/particles/particleInertial.c:198]: If you get a chance, would you mind removing the extra spaces in the files?

File: ablateCore/particles/particles.c [hunk 2]
```diff
@@ -163,17 +167,18 @@ PetscErrorCode ParticleView(ParticleData particleData, PetscViewer viewer) {
     PetscFunctionReturn(0);
 }
 
-PetscErrorCode ParticleViewFromOptions(ParticleData particleData, char *optionname) {
+PetscErrorCode ParticleViewFromOptions(ParticleData particleData,PetscObject obj, char *title) {
     PetscFunctionBegin;
+    Vec            particleVector;
     PetscErrorCode ierr;
-    PetscViewer viewer;
```
Comment (mmcgurn) [inline ablateCore/particles/particles.c:169]: I think that you are introducing old code here.  I think that putting back this old code may result in some bugs.

**Review by mmcgurn — CHANGES_REQUESTED** [review]
@pakserep Thanks for putting in this PR! and all of the hard work that goes into it.  I think that there is a bug in replacing newer code and you may need to rebase your code, but overall it looks good!  Would you also mind adding a little documentation about how particle system/equations? I'd be happy to help.

**Comment by pakserep** [comment]
hi @mmcgurn, thanks for taking the time and reviewing the PR and thanks for the comments. I rebased my feature branch (pakserep/inertialParticle-bug) onto the updated main branch (upstream) and pushed my changes from cloned to my fork. it sees the PR is updated and reflecting the new commits. Please let me know if this resolves the issues. Once this is resolved I will do another PR for documentation, if that's ok. Thanks.

**Review by mmcgurn — APPROVED** [review]

**Comment by mmcgurn** [comment]
@pakserep You should now be good to squash and merge.  Make sure you select Squash and Merge option.

**Pull request merged**